source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -337,6 +490,50 @@ void SplitKeysSumResultRequest::onChildResponse(Common::Redis::RespValuePtr&& va
   }
 }
 
+bool SplitKeysSumResultRequest::onChildMovedRedirection(Common::Redis::RespValue& value,
+                                                        uint32_t index) {
+  std::vector<absl::string_view> err;
+  if (redirectionArgsInvalid(incoming_request_, value, err)) {
+    return false;
+  }
+
+  std::vector<Common::Redis::RespValue> values(2);
+  values[0].type(Common::Redis::RespType::BulkString);
+  values[0].asString() = incoming_request_->asArray()[0].asString();
+  values[1].type(Common::Redis::RespType::BulkString);
+  values[1].asString() = incoming_request_->asArray()[index + 1].asString();
+  Common::Redis::RespValue single_fragment;
+  single_fragment.type(Common::Redis::RespType::Array);
+  single_fragment.asArray().swap(values);
+
+  this->pending_requests_[index].handle_ = conn_pool_.makeRequestToHost(
+      std::string(err[2]), single_fragment, this->pending_requests_[index]);
+  return (this->pending_requests_[index].handle_ != nullptr);
+}
+
+bool SplitKeysSumResultRequest::onChildAskRedirection(Common::Redis::RespValue& value,
+                                                      uint32_t index) {
+  std::vector<absl::string_view> err;
+  if (redirectionArgsInvalid(incoming_request_, value, err)) {
+    return false;
+  }
+
+  std::vector<Common::Redis::RespValue> values(3);
+  values[0].type(Common::Redis::RespType::BulkString);
+  values[0].asString() = "asking";
```

Comment:
This pattern/code seems repeated a ton. Is there any way to move this into a helper function that is shared?

Reply:
I'm creating a recreateRequest() method for each request type. It'll be up soon.

Reply:
I've created a new recreate() method for each request type to reduce the duplication of code.